Addressed many of the changes from Harlan's PR.

bmbowdish opened the pull request:
* refactored parse* to be parseResource 
  * uses generics
* removed simplePrint() for CustomStringConvertible
* Created Result<Value> enum to handle what parseResource returns
* turned some [type?] to [type]? or [type] 
* each get* now throws if it fails

bmbowdish committed:
Addressed many of the changes from Harlan's PR.

bmbowdish committed:
Fixed some structs description

bmbowdish commented:
Thanks again to #23

bmbowdish committed:
Fixed some spacing issues.

liam-middlebrook commented:
This isn’t a code specific issue, and it’s not something that you *have* to do but I’ve found this blog post to be helpful in the past to understand a common methodology for writing commit messages. 

https://chris.beams.io/posts/git-commit/

bmbowdish commented:
Summarize changes in around 50 characters or less

More detailed explanatory text, if necessary. Wrap it to about 72
characters or so. In some contexts, the first line is treated as the
subject of the commit and the rest of the text as the body. The
blank line separating the summary from the body is critical (unless
you omit the body entirely); various tools like `log`, `shortlog`
and `rebase` can get confused if you run the two together.

Explain the problem that this commit is solving. Focus on why you
are making this change as opposed to how (the code explains that).
Are there side effects or other unintuitive consequences of this
change? Here's the place to explain them.

Further paragraphs come after blank lines.

 - Bullet points are okay, too

 - Typically a hyphen or asterisk is used for the bullet, preceded
   by a single space, with blank lines in between, but conventions
   vary here

If you use an issue tracker, put references to them at the bottom,
like this:

Resolves: #123
See also: #456, #789

--------

I thought this was especially useful.

bmbowdish committed:
Reflects Changes 

Harlan suggested some changes and I implemented them. This README now reflects that.

bmbowdish committed:
Merge remote-tracking branch 'origin/master' into refactParsing

bmbowdish committed:
Merge branch 'refactParsing' of https://github.com/bmbowdish/Swiftfall into refactParsing
